provision/minikube/keycloak/templates/keycloak.yaml

```diff
@@ -148,6 +148,18 @@ spec:
             imagePullPolicy: Never
 {{ end }}
             env:
+              # Limiting Keycloak thread pool as it might lead to an exhaustion of the JGroups thread pool.
+              # The combined worker thread pool size of all Keycloak instances shouldn't exceed the size of the JGroups thread pool on each node.
```

Comment:
I am not following the reasoning behind `all Keycloak instances shouldn't exceed the size of the JGroups thread pool`. Isn't JGroups thread pool configured per pod? Based on that reasoning, shouldn't we set `QUARKUS_THREAD_POOL_MAX_THREADS` to `100`?

Reply:
As I understand it, the Keycloak worker threads use JGroups to connect to other Keycloak instances to access the cache there. Which node they connect to depends on where the session is stored. It might be that at a given moment in time all nodes connect with all their threads to the same Keycloak node, and even then this shouldn't exceed the number of JGroup threads on that node (this is my understanding). 

In practice, we've seen that 200 threads on 6 Keyloak Pods lead to a JGroups thread pool of 100 being exhausted in several occasions. This PR will use 16 Keycloak threads, that the sum of all Keycloak threads will never exceed 100 JGroups threads. Let's see how the system behaves, maybe the optimal values will be somewhere in the middle, as some threads will use local data only (about 33% of them). Let's use this PR and see once the numbers come in.

Reply:
Thanks for the explanation @ahus1, makes sense to me.

Reply:
From yesterday's chaos testing: With 16 worker threads on 6 pods, the JGroup worker pool was at around 60 active threads which is well below the maximum of 100 which could cause problems. It also shows that while the number of workers could be a bit higher, but not much. Keeping it at this number for now in the PR.